### ISSUE TITLE
Cuda OOM even with batch size 1 and lower seq length on 16GB GPU

### ISSUE THREAD

**Issue by noobmldude** [loubnabnl/santacoder-finetuning#16]
I tried the finetuning script on a single V100 GPU with 16GB GPU Memory and > 200GB VRAM.
I still get CUDA OOM.

Should it be possible to finetune on a single V100 GPU?
Am I doing something wrong? 
Any tricks to get it running is very much appreciated.

**Comment by loubnabnl** [comment]
Can you make sure you have gradient checkpointing turned on? You can also use half precision or reduce the context length to 1024 instead of 2048. You can check this  [Google Colab T4](https://colab.research.google.com/drive/1UMjeXHwOldpLnWjdm1499o2IYy0RgeTw?usp=sharing) for SantaCoder finetuning where it fits in a T4

**Comment by bowencarry** [comment]
I have the same problem, did you solve it？

**Comment by loubnabnl** [comment]
I recommend checking this code, which uses quantization and PEFT to reduce the memory footprint: https://github.com/bigcode-project/starcoder2/blob/main/finetune.py
